src/deluge/gui/menu_item/enumeration.cpp

```diff
@@ -13,48 +13,25 @@ void Enumeration::beginSession(MenuItem* navigatedBackwardFrom) {
 }
 
 void Enumeration::selectEncoderAction(int32_t offset) {
-	this->setValue(this->getValue() + offset);
 	int32_t numOptions = size();
-	int32_t sign = (offset < 0) ? -1 : ((offset > 0) ? 1 : 0);
-
-	switch (numOptions) {
-	case 0:
-		[[fallthrough]];
-	case 1:
-		[[fallthrough]];
-	case 2:
-		offset = 1 * sign;
-		break;
-	case 3:
-		offset = std::min<int32_t>(offset * sign, 2) * sign;
-		break;
-	case 4:
-		offset = std::min<int32_t>(offset * sign, 3) * sign;
-		break;
-	case 5:
-		offset = std::min<int32_t>(offset * sign, 4) * sign;
-		break;
-	default:
-		break;
-	}
```

Comment:
I think we still need this offset clamp logic, otherwise small menus can end up being unnavigable with the shift+select scroll shortcut.

Reply:
Maybe we should do something alternative - shift + select offsets by 5 right? Scrolling by n-1 on a short list of n when wrapping is confusing. If I have 1,2,3,4, I'm on 2, and I shift scroll forward, the net effect is scrolling backwards by 1. How about if n <= 5, we just make shift scroll jump to the beginning or end of the list? On OLED with no wrapping, this is the behavior you'd get anyway.